examples/ex7_1.ipynb

```diff
@@ -357,7 +357,7 @@
     "for row in kg.query(sparql):\n",
     "    p1 = subgraph.transform(row.p1)\n",
     "    p2 = subgraph.transform(row.p2)\n",
-    "    psl.add_data_row(\"Likes\", [p1, p2])"
+    "    psl.add_data_row(\"Likes\", [p1, p2])  # anchoring PR discussion"
```

Comment:
this notebook is really fascinating.
I'm not sure about one thing though:
the rule we added is 
```python
psl.add_rule("Likes(P1, L) & Likes(P2, L) & (P1 != P2) -> Knows(P1, P2)", weight=10.0, squared=True)
```
but the observations we add are in effect `Likes(P1, P2)` no? 
not sure at all about what I'm saying. It's the second time I hear about PSL (first time being your talk 😂 )

Reply:
The PSL example was difficult to deconstruct from their docs + code.  FWIW, I've taught a graph course before with one of the authors – Matthias Broecheler who's now at DataStax – plus I'm a big fan of Lise Getoor's work making _statistical relational learning_ more practical for industry use cases.  I was trying to show how one of their simpler examples would integrate with an RDF graph, since the PSL rules can be really useful for data quality checks on graphs in general.

I'll double-check the way that those rules' observations and grounded values were loaded from the KG. The results _seemed_ to agree 😅  Will check -- I added that code way too quickly. FWIW, their lab at UCSC is really good at proving math theorems about convex optimization methods, although I find it hard to link up the proofs in their papers with their code + docs.

So what I should've done is set up some unit test based on their expected output values.

The code here:
```
psl.add_data_row("Likes", [p1, p2])
```
is probably incorrect. I'll need to check through UCSC's data files.